js/src/carousel.js

```diff
@@ -219,7 +218,8 @@ class Carousel extends BaseComponent {
   _getConfig(config) {
     config = {
       ...Default,
-      ...config
+      ...Manipulator.getDataAttributes(this._element),
+      ...(typeof config === 'object' ? config : {})
```

Comment:
This check can be omitted here since you have already placed this in the definition of `getOrCreateInstance`?

Reply:
And in case the Component is initialized like: `new Alert(elem, config)`?

If we streamline `_getConfig`, it can be omitted on `getOrCreateInstance`  I suppose

Reply:
Yes @GeoSot, I am also pointing the same thing. You can remove `typeof config === 'object' ? config : {}` from the `_getConfig` method IMO. :thinking:

Reply:
and where are we  going to check it? :thinking:

Reply:
Okay NVM, if you are checking the configuration for the initialization of the plugin then it's good.